### PR TITLE
Incorporate RxJS bundles

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1139,7 +1139,35 @@ gulp.task('!bundles.js.umd', ['build.js.dev'], function() {
       resolve: resolveOptions(devOrProd),
       module: {preLoaders: [{test: /\.js$/, loader: 'source-map-loader'}]},
       devtool: devOrProd === 'dev' ? 'inline-source-map' : undefined,
-      output: outputOptions(outFileName, devOrProd)
+      output: outputOptions(outFileName, devOrProd),
+      externals: {
+        'rxjs/Observable': 'umd Rx',
+        'rxjs/Subject': 'umd Rx',
+        'rxjs/subject/ReplaySubject': {
```
Comment (jeffbcross) [inline gulpfile.js]: Actually, this could be simplified since `ReplaySubject` is on the `Rx` object.

File: gulpfile.js [hunk 2]
```diff
@@ -1139,7 +1139,35 @@ gulp.task('!bundles.js.umd', ['build.js.dev'], function() {
...
+          commonjs: 'rxjs/subject/ReplaySubject',
+          commonjs2: 'rxjs/subject/ReplaySubject',
+          amd: 'rxjs/subject/ReplaySubject',
+          root: ['Rx']
+        },
+        'rxjs/operator/take': {
```
Comment (jeffbcross) [inline gulpfile.js]: These have to be special-cased because they're not directly available on the `Rx` object.

File: gulpfile.js [hunk 3]
```diff
@@ -1139,7 +1139,35 @@ gulp.task('!bundles.js.umd', ['build.js.dev'], function() {
...
+          commonjs: 'rxjs/operator/take',
+          commonjs2: 'rxjs/operator/take',
```
Comment (jeffbcross) [inline gulpfile.js]: Hmm, actually, I bet these will not work with the current UMD from rxjs because it is all relative paths internally.

**Comment by jeffbcross** [comment]
CC @IgorMinar @vsavkin @robwormald

**Comment by IgorMinar** [comment]
I don't see the shrinkwrap clean file being updated

**Comment by jeffbcross** [comment]
@IgorMinar I pushed the cleaned shrinkwrap (FYI travis was green before I force pushed)

**Comment by IgorMinar** [comment]
@jeffbcross it seems that you forgot to run `./tools/npm/clean-shrinkwarp.js` - I'll fix the PR and merge it for you. next time please follow: https://github.com/angular/angular/blob/master/npm-shrinkwrap.readme.md

**Comment by IgorMinar** [comment]
oh.. nevermind. I guess I was looking at a stale version of the PR. npm-shrinkwrap.clean.json is updated

**Comment by jeffbcross** [comment]
> User @IgorMinar does not have PR merging privileges.

I guess @mary-poppins has become self-aware and is taking over as angular tech lead.

**Pull request closed without merging**